pallets/dday-protocol/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+        StorageMap<_, Blake2_128Concat, NFTId, TransmissionData<T::AccountId>, OptionQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig {}
+
+    #[cfg(feature = "std")]
+    impl Default for GenesisConfig {
+        fn default() -> Self {
+            Self {}
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {}
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        // Transmission has been store
+        TransmissionStored,
+        // Transmission has been completed
+        TransmissionCompleted,
+        // Transmission has been cancelled
+        TransmissionCancelled,
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        // This function is reserved to the owner of a capsule.
+        NotCapsuleOwner,
+        // Unknown Capsule
+        UnknownCapsule,
+        // Is already stored as transmission !
+        AlreadyInTransmission,
+        // Should be stored as transmission !
+        NotInTransmission,
+        // Should be stored as transmission !
+        IsListedForSale,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(
+            n: <T as frame_system::Config>::BlockNumber,
+        ) -> frame_support::weights::Weight {
+            for data in Transmissions::<T>::iter() {
+                let (nft_id, transmission_data) = data;
+                if n >= transmission_data.delivery_date.into() {
+                    if Self::transmits_capsule(nft_id, transmission_data).is_ok() {
+                        Self::deposit_event(Event::TransmissionCompleted);
+                    } else {
+                        return 0;
+                    }
+                }
+            }
+            1
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        // Add a new tranfer into storage
+        #[pallet::weight(T::WeightInfo::dday_transmission())]
+        pub fn dday_transmission(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            recipient: T::AccountId,
+            delivery_date: BlockNumber,
+        ) -> DispatchResult {
+            let account_id = ensure_signed(origin)?;
+            let capsule = T::Capsules::get_capsule(nft_id).ok_or(Error::<T>::UnknownCapsule)?;
+            ensure!(account_id == capsule.owner, Error::<T>::NotCapsuleOwner);
+
+            ensure!(
+                T::NFTs::is_listed_for_sale(nft_id) == Some(false),
+                Error::<T>::IsListedForSale
+            );
+            ensure!(
+                T::NFTs::is_in_transmission(nft_id) == Some(false),
+                Error::<T>::AlreadyInTransmission
+            );
+
+            Transmissions::<T>::insert(nft_id, TransmissionData::new(recipient, delivery_date));
+            T::NFTs::set_in_transmission(nft_id, true)?;
```

Comment:
Invert those two lines: put the one that may fail before the other.
And in a spirit of safeness make this extrinsic #[transactional]
Same for `cancel`